File: khsexk-Java/src/PrefixSum/BOJ_1806.java
package PrefixSum;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class BOJ_1806 {
	static int N, S;
	static int[] prefix;
	static int min = Integer.MAX_VALUE;
	
	public static void main(String[] args) throws IOException{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
		int[] input = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
		N = input[0];
		S = input[1];
		
		StringTokenizer st = new StringTokenizer(br.readLine());
		prefix = new int[N+1];
		
		for(int i=0; i<N ; i++) {
			prefix[i] = Integer.parseInt(st.nextToken());
		}
		
		int start = 0, end = 0; 
		int sum = 0;
		while(start <= end && end <= N) {
			if(sum < S)
				sum += prefix[end++];
			else {
				min = Math.min(min, end-start);
				sum -= prefix[start++];
			}
		}
		
		if(min == Integer.MAX_VALUE)
       		System.out.println(0);        
        else
       		System.out.println(min);
	}

}

File: yongshub-python/약점체크/BOJ_1806.py
import sys
input = sys.stdin.readline
N, S = map(int, input().split())
arr = list(map(int, input().split()))
min_value = 100000001
start, end = 0, 0
sum_value = 0
while True:
    if sum_value < S:
        if end == N:
            break
        sum_value += arr[end]
        end += 1
    else:
        if end == N + 1:
            break
        elif sum_value - arr[start] >= S:
            min_value = min(min_value, end - (start + 1))
        else:
            min_value = min(min_value, end - start)
        sum_value -= arr[start]
        start += 1
        if min_value == 1: break
print(0) if min_value == 100000001 else print(min_value)
